sos/report/plugins/openstack_placement.py

```diff
@@ -39,6 +40,38 @@ def setup(self):
                 suggest_filename="placement-manage_db_version"
             )
 
+            vars_all = [p in os.environ for p in [
+                        'OS_USERNAME', 'OS_PASSWORD']]
+
+            vars_any = [p in os.environ for p in [
+                        'OS_TENANT_NAME', 'OS_PROJECT_NAME']]
+
+            if not (all(vars_all) and any(vars_any)):
+                self.soslog.warning("Not all environment variables set. "
+                                    "Source the environment file for the user "
+                                    "intended to connect to the OpenStack "
+                                    "environment.")
+            else:
+                self.add_cmd_output("openstack resource provider list")
+
+                cmd = "openstack resource provider list -f value"
+                res = self.exec_cmd(cmd)
+                if res['status'] == 0:
+                    resource_provider_list = res['output']
+                    for provider in resource_provider_list.splitlines():
+                        provider = provider.split()[0]
+                        sub_cmds = [
+                            "inventory",
+                            "trait",
+                            "aggregate",
+                        ]
+                        for sub_cmd in sub_cmds:
+                            cmd = (f"openstack resource provider {sub_cmd} "
+                                   " list {provider}")
+                            file = (f"resource_provider_{sub_cmd}-{provider}"
+                                    ".log")
+                            self.add_cmd_output(cmd, suggest_filename=file)
```

Comment:
Same, I don't see the need for renaming the command output to a log file.

This can also be neatly buttoned up as

```python
self.add_cmd_output([
    f"openstack resource provider {sub_cmd} list {provider}"
    for sub_cmd in sub_cmds
])
```